File: app/javascript/channels/answers_channel.js
import consumer from "./consumer"

$(document).on('turbolinks:load', function (){
  console.log('Controller Name:', gon.controller_name);
  console.log('Controller Action:', gon.controller_action);
  const answersContainer = $('.answers');
  if (answersContainer.length > 0) {
    const currentUser = answersContainer.data('current-user');
    console.log('Current User:', currentUser);

  if((gon.controller_name === 'questions') && (gon.controller_action === 'show')){
    consumer.subscriptions.create({
      channel:"AnswersChannel",
      id: gon.params_id}  , 
    
    {
      connected() {
        // Called when the subscription is ready for use on the server
        console.log('CONNECTED ANSWER!');
      
      },
    
      disconnected() {
        // Called when the subscription has been terminated by the server   
      },
    
      received(data) {
        // Called when there's incoming data on the websocket for this channel
        console.log(data)
        $('.answers').append(data.answer)
      }
    });
  };

}
  
});

File: app/channels/answers_channel.rb
class AnswersChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    question = Question.find(params[:id])
    stream_from "details_data_for_question_#{question.id}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  
  def receive(data)
    current_user = User.find(data['current_user_id'])
    # Теперь у вас есть текущий пользователь для этого подключения
  end
 
end
